packages/main/src/plugin/kubernetes/kubernetes-client.ts

```diff
@@ -1264,6 +1264,209 @@ export class KubernetesClient {
     return this.syncResources(context, manifests, 'apply', namespace);
   }
 
+  /**
+   * Parse a kubeconfig file and extract contexts information
+   *
+   * @param filePath path to the kubeconfig file
+   * @returns array of KubeContext objects found in the file
+   */
+  async parseKubeconfigFile(filePath: string): Promise<KubeContext[]> {
+    // Check if file exists
+    if (!fs.existsSync(filePath)) {
+      throw new Error(`Kubeconfig file ${filePath} does not exist`);
+    }
+
+    try {
+      // Create a temporary KubeConfig instance to parse the file
+      const tempConfig = new KubeConfig();
+      tempConfig.loadFromFile(filePath);
+
+      const contexts: KubeContext[] = [];
+      const currentContextName = tempConfig.getCurrentContext();
+
+      // Process each context
+      tempConfig.contexts.forEach(context => {
+        // Find the cluster information
+        const cluster = tempConfig.clusters.find(c => c.name === context.cluster);
+
+        contexts.push({
+          name: context.name,
+          cluster: context.cluster,
+          user: context.user,
+          namespace: context.namespace,
+          currentContext: context.name === currentContextName,
+          clusterInfo: cluster
+            ? {
+                name: cluster.name,
+                server: cluster.server,
+                skipTLSVerify: cluster.skipTLSVerify,
+                tlsServerName: cluster.tlsServerName,
+              }
+            : undefined,
+        });
+      });
```

Comment:
an extracted method contextToKubeContext (or equivalent name) would be more readable